include/adm/serial/transport_track_format.hpp

```diff
@@ -0,0 +1,121 @@
+   * +======================+============================+==============================+
+   * | transportID          | :class:`TransportId`       | :class:`RequiredParameter`   |
+   * +----------------------+----------------------------+------------------------------+
+   * | transportName        | :type:`TransportName`      | :class:`OptionalParameter`   |
+   * +----------------------+----------------------------+------------------------------+
+   * | numTracks            | :type:`NumTracks`          | :class:`OptionalParameter`   |
+   * +----------------------+----------------------------+------------------------------+
+   * | numIDs               | :type:`NumIds`             | :class:`OptionalParameter`   |
+   * +----------------------+----------------------------+------------------------------+
+   * | audioTrack           | :type:`AudioTracks`        | :class:`VectorParameter`     |
+   * +----------------------+----------------------------+------------------------------+
+   * \endrst
+   */
+  class TransportTrackFormat
+      : private detail::TransportTrackFormatBase,
+        private detail::AddWrapperMethods<TransportTrackFormat> {
+   public:
+    using tag = TransportTrackFormatTag;
+    /// Type that holds the id for this element;
+    using id_type = TransportId;
+
+    /**
+     * @brief Constructor template
+     *
+     * Templated constructor which accepts a variable number of ADM parameters
+     * in random order after the mandatory ADM parameters.
+     */
+    template <typename... Parameters>
+    explicit TransportTrackFormat(TransportId id,
+                                  Parameters... optionalNamedArgs);
+
+    using detail::AddWrapperMethods<TransportTrackFormat>::has;
+    using detail::AddWrapperMethods<TransportTrackFormat>::get;
+    using detail::AddWrapperMethods<TransportTrackFormat>::unset;
+    using detail::AddWrapperMethods<TransportTrackFormat>::isDefault;
+    using detail::TransportTrackFormatBase::add;
+    using detail::TransportTrackFormatBase::get;
+    using detail::TransportTrackFormatBase::has;
+    using detail::TransportTrackFormatBase::isDefault;
+    using detail::TransportTrackFormatBase::remove;
+    using detail::TransportTrackFormatBase::set;
+    using detail::TransportTrackFormatBase::unset;
+
+    /**
+     * @brief Print overview to ostream
+     */
+    void print(std::ostream& os) const;
+    friend class detail::AddWrapperMethods<TransportTrackFormat>;
+  };
+
+  inline bool operator==(TransportTrackFormat const& lhs,
+                         TransportTrackFormat const& rhs) {
+    auto result = lhs.get<TransportId>() == rhs.get<TransportId>();
+    result = result && lhs.get<TransportName>() == rhs.get<TransportName>();
+    result = result && lhs.get<NumTracks>() == rhs.get<NumTracks>();
+    result = result && lhs.get<NumIds>() == rhs.get<NumIds>();
+    result = result && lhs.get<AudioTracks>() == rhs.get<AudioTracks>();
+    return result;
```

Comment:
should use `OptionalComparator`

Reply:
0c2051e4753b11dc72ddf7256939a63b8df2f4a9